### PR TITLE
Classic page object

### PR DESCRIPTION
**Pull request by VitalyApanovich**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-server</artifactId>
+            <version>3.0.1</version>
```
Comment (EugenBorisik) [inline pom.xml]: Use the latest version of selenium webdriver api (3.14.0)

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>6.11</version>
+        </dependency>
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-api</artifactId>
+            <version>3.0.1</version>
```
Comment (EugenBorisik) [inline pom.xml]: Use the latest version of selenium webdriver api (3.14.0)

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-remote-driver</artifactId>
+            <version>3.0.1</version>
```
Comment (EugenBorisik) [inline pom.xml]: Use the latest version of selenium webdriver api (3.14.0)

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-chrome-driver</artifactId>
+            <version>3.0.1</version>
```
Comment (EugenBorisik) [inline pom.xml]: Use the latest version of selenium webdriver api (3.14.0)

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-firefox-driver</artifactId>
+            <version>3.0.1</version>
```
Comment (EugenBorisik) [inline pom.xml]: Use the latest version of selenium webdriver api (3.14.0)

File: src/main/java/Driver.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+            WebDriver d = new ChromeDriver(caps);
+            d.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+            d.manage().window().maximize();
+            driver.set(d);
+        }
+        return driver;
+    }
+
+    static void openLoginPage() {
+            getDriver().get().navigate().to(URL);
```
Comment (EugenBorisik) [inline src/main/java/Driver.java]: Reformat code, too big indent

File: src/main/java/InboxPage.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+class InboxPage extends Page{
```
Comment (EugenBorisik) [inline src/main/java/InboxPage.java]: Add single whitespace after Page

File: src/main/java/InboxPage.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    InboxPage() {
+        super(TITLE);
+    }
+
+    LoginPage logout() {
+        getLogoutLink().click();
+        return new LoginPage();
+    }
+
+    WebElement getLogoutLink() {
```
Comment (EugenBorisik) [inline src/main/java/InboxPage.java]: It is strange that you are returning WebElement and method is not private. Page Object should encapsulate everything connected to WebElement. Please update

File: src/main/java/LoginPage.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+class LoginPage extends Page{
```
Comment (EugenBorisik) [inline src/main/java/LoginPage.java]: Add whitespace after Page

File: src/main/java/LoginPage.java [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    private final static String TITLE = "Mail.Ru: почта, поиск в интернете, новости, игры";
```
Comment (EugenBorisik) [inline src/main/java/LoginPage.java]: static final

File: src/main/java/LoginPage.java [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+    InboxPage login(String login, String pass) {
+        driver.findElement(USERNAME_INPUT).sendKeys(login);
+        driver.findElement(PASSWORD_INPUT).sendKeys(pass);
+        getLoginButton().click();
+        Waiter.explicitWait(20);
+        return new InboxPage();
+    }
+
+    WebElement getLoginButton() {
```
Comment (EugenBorisik) [inline src/main/java/LoginPage.java]: See comment in InboxPage class

File: src/main/java/LoginPage.java [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+    LoginPage() {
+        super(TITLE);
+    }
+
+    InboxPage login(String login, String pass) {
+        driver.findElement(USERNAME_INPUT).sendKeys(login);
+        driver.findElement(PASSWORD_INPUT).sendKeys(pass);
+        getLoginButton().click();
+        Waiter.explicitWait(20);
```
Comment (EugenBorisik) [inline src/main/java/LoginPage.java]: See comment in Waiter class

File: src/main/java/Waiter.java [hunk 13]
```diff
@@ -0,0 +1,8 @@
+import java.util.concurrent.TimeUnit;
+
+public class Waiter {
+
+    public static void explicitWait(int waitTime) {
```
Comment (EugenBorisik) [inline src/main/java/Waiter.java]: It is not explicit wait. Explicit wait refers to visibility/text appearance/etc. Please update this method to be real explicit waiter and then use it where needed

**Review by EugenBorisik — CHANGES_REQUESTED** [review]
@VitalyApanovich please update code

File: src/main/java/Waiter.java (hunk 13 above)
Comment (VitalyApanovich) [reply src/main/java/Waiter.java]: renaming to implicitWait as I am not waiting for any element on the Page. Mail.ru is just a crazy page for loading and don't have enough time to track which elements I should wait for. So need just configurable method for implicit wait.

Comment (EugenBorisik) [reply src/main/java/Waiter.java]: Okay, just update implicit waiter back to default value after that specific place, please.

**Review by EugenBorisik — APPROVED** [review]

**Pull request merged**